src/inventory_ui.cpp

```diff
@@ -271,23 +271,21 @@ static const selection_column_preset selection_preset{};
 
 bool inventory_entry::is_hidden() const
 {
-    if( !is_item() ) {
+    // non-items and entries not added recursively (from a container) can't be hidden
+    if( !is_item() || topmost_parent == nullptr ) {
         return false;
     }
-    item_location it = locations.front();
-    bool hidden = false;
-    if( topmost_parent != nullptr ) {
-        while( it.has_parent() ) {
-            item_location const prnt = it.parent_item();
-            hidden |= prnt.get_item()->contained_where( *it )->settings.is_collapsed();
-            if( prnt.get_item() == topmost_parent ) {
-                break;
-            }
-            it = prnt;
+
+    item_location item = locations.front();
+    while( item.has_parent() ) {
+        item_location parent = item.parent_item();
+        if( parent.get_item()->contained_where( *item )->settings.is_collapsed() ) {
+            return true;
```

Comment:
This breaks uncollapsing items inside the "Examine contents" window (`o` by default)

Reply:
My mistake, didn't know about inventory examiner. Should be fixed now. Also, the is_hidden() cleanup is non-essential to this bugfix and can be removed if you want. 

Just saw it along the way when trying to debug the reflow issue and thought it'd be more efficient to shortcircuit the 'or' once it found a parent that was collapsed.

Reply:
In general you shouldn't restyle code unless you're fixing bugs in it. But this bit looks much better now so I think it's OK.